### DISCUSSION TITLE
Rectangle Pro - Can't seem to get screen edge gaps to work on second monitor (screenEdgeGapTop)

### DISCUSSION THREAD

**Discussion by Dolvur** [rxhanson/RectanglePro-Community#433] [Q&A]
I am using a sketchybar as a custom bar for showing my spaces and what is on them (while hiding the original bar)

![image](https://github.com/rxhanson/Rectangle/assets/69213759/2eba76b7-20f9-441d-9f3f-1ac409ab771a)

However, I cannot seem to be set a screen gap for the top portion of the screen so I can see the bar when I snap the windows. I tried running `defaults write com.knollsoft.Rectangle screenEdgeGapTop -int 32` but it still does not work on my second monitor.

The strange thing is that it works on the laptop screen, even without the screenEdgeGapTop set.

`defaults read com.knollsoft.Rectangle` gives me:

```
{
    SUEnableAutomaticChecks = 0;
    SUHasLaunchedBefore = 1;
    alternateDefaultShortcuts = 1;
    lastVersion = 81;
    reflowTodo =     {
        keyCode = 45;
        modifierFlags = 786432;
    };
    screenEdgeGapTop = 32;
    screenEdgeGapsOnMainScreenOnly = 0;
    subsequentExecutionMode = 1;
    toggleTodo =     {
        keyCode = 11;
        modifierFlags = 786432;
    };
}
```
I am using Rectangle Pro

**Comment by Dolvur** [comment]
Nvm, I managed to fix it. With rectangle pro it seems like I should write to `com.knollsoft.Hookshot`. After doing that and restarting Rectangle Pro, it worked

**Reply by Dolvur** [reply]
Now, the upper gap on the laptop screen are too big. Even after I removed the corresponding options from `com.knollsoft.Rectangle` 🙃

`com.knollsoft.Hookshot`:
```
{
    SUEnableAutomaticChecks = 0;
    SUHasLaunchedBefore = 1;
    adjacentResizing = 1;
    allowAnyShortcut = 1;
    almostMaximize =     {
    };
    analogAdjacentResizing = 1;
    appLeftHalf =     {
    };
    applyGapToEdges = 1;
    bottomCenterLeftEighth =     {
    };
    bottomCenterSixth =     {
    };
    bottomLeftEighth =     {
    };
    bottomLeftSixth =     {
    };
    bottomLeftThird =     {
    };
    bottomRightNinth =     {
    };
    bottomRightSixth =     {
    };
    bottomRightThird =     {
    };
    centerTwoThirds =     {
    };
    disabledApps = "[]";
    doubleClickTitleBar = 0;
    fillBottomLeft =     {
    };
    fillBottomRight =     {
    };
    fillLeft =     {
        keyCode = 4;
        modifierFlags = 786432;
    };
    fillRight =     {
        keyCode = 37;
        modifierFlags = 786432;
    };
    fillTopLeft =     {
    };
    fillTopRight =     {
    };
    firstFourth =     {
    };
    firstSixth =     {
    };
    firstThreeFourths =     {
    };
    gapSize = 0;
    gestures = "[33,{\\"flags\\":1441792,\\"trail\\":0}]";
    landscapeSnapAreas = "[2,{\\"action\\":2},6,{\\"action\\":13},8,{\\"action\\":14},4,{\\"compound\\":-2},5,{\\"compound\\":-3},7,{\\"compound\\":-4},3,{\\"action\\":16},1,{\\"action\\":15}]";
    lastFourth =     {
    };
    lastThreeFourths =     {
    };
    lastVersion = 157;
    launchOnLogin = 1;
    longReticleSpec = "{\\"l\\":30,\\"id\\":1,\\"r\\":31}";
    mainReticleSpec = "{\\"b\\":29,\\"r\\":1,\\"tl\\":15,\\"t\\":2,\\"id\\":0,\\"tr\\":16,\\"bl\\":13,\\"l\\":0,\\"br\\":14}";
    moveLeft =     {
    };
    moveUp =     {
    };
    nextDisplayRatio =     {
    };
    nextSpace =     {
    };
    nudgeDown =     {
    };
    nudgeUp =     {
    };
    prevDisplayRatio =     {
    };
    prevSpace =     {
    };
    quickActions = 2;
    screenEdgeGapTop = 32;
    secondFourth =     {
    };
    stashEnabled = 2;
    thirdFourth =     {
    };
    throwScrollResize = 1;
    tidy =     {
    };
    tile2x3 =     {
    };
    todo = 2;
    todoApplication = "com.apple.Preview";
    todoMode = 0;
    topCenterLeftEighth =     {
    };
    topCenterSixth =     {
    };
    topLeftEighth =     {
    };
    topLeftSixth =     {
    };
    topLeftThird =     {
    };
    topRightSixth =     {
    };
    topRightThird =     {
    };
    winModFlags = 0;
}
```

**Reply by rxhanson** [reply]
Unless I am misunderstanding, you'll just want to play around with that value (32) to come up with an optimal one?

**Reply by Dolvur** [reply]
Sorry if I was a little bit confusing, the initial problem of not being able to add gaps to the top was solved by setting the `screenEdgeGapTop = 32;` value in `com.knollsoft.Hookshot`. The gaps are now perfect on my external monitor.

However, my current issue is that the laptop screen originally (for some reason) had a perfect gap size at the top beforehand. So now when I set the value of `screenEdgeGapTop=32;` I get a gap size that is twice as big.

I am not sure this is an issue with rectangle though, since even with rectangle and sketchybar disabled, if I double click a window to maximise it, macOS seems to still keep a gap at the top.

One solution I found was to set `screenEdgeGapsOnMainScreenOnly` to `true`. But I still have the issue with twice the gap size when I am only using the laptop

**Reply by rxhanson** [reply]
Hmm. This sounds like it's external to Rectangle/Pro, since the app is going to use whatever the OS says the dimensions of the screen are, and apply the gaps as configured. If more details emerge or it does seem like a bug on Rectangle/Pro's side, let me know.

**Reply by JDuffe** [reply]
Hi @Dolvur, I am currently configuring my sketchybar (also using Rectangle Pro as my wm) and ran into the exact same problem as you described (works on external screen but twice the gap on internal screen etc.). 
Did you find or come up with a fix for that since then?

**Reply by Dolvur** [reply]
@JDuffe, unfortunately not. I went back to using native macos bar instead of sketchybar, so I stopped looking into fixes

**Reply by JDuffe** [reply]
Ok, thanks anyway. If I do come up with a solution, I'll post it here!
